### ISSUE TITLE
Intellisense / project code analysis feature

### ISSUE THREAD

**Issue by pwFoo** [pizzapanther/Super-Neutron-Drive#73]
Is a code analysis feature planned like static code analysis to get improved autocomplete for cms / cmf classes and methods (php)?

**Comment by pizzapanther** [comment]
Some languages have this already like javascript but php does not yet.

Since Google announced the deprecation Chrome Apps, the conversion of Neutron to a web based app is higher priority and so I probably won't add this feature in this version. However, I'll bring this feature request to the next version.

**Comment by pwFoo** [comment]
I compare Neutron Drive with [Caret](https://chrome.google.com/webstore/detail/caret/fljalecfjciodhpcledpamjachpmelml?hl=de) at the moment, which is a Chrome App too. I like Caret UI (clear, similar features, language / theme select.

Remove the chrome dependency (which I use as prefered browser) and just use it with any browser web based would be nice. 👍

**Comment by pizzapanther** [comment]
Yes chrome dependency will be removed but you'll still need an extension if you want to do things like local file editing. So I think there will be a chrome extension and possible a firefox extension since I believe firefox has the same extension APIs now.

**Comment by pwFoo** [comment]
So a full browser based solution like [Cloud9](https://c9.io/) or [Orion](https://orionhub.org) isn't planned? I don't know how it's done or how much additional work it would be. Just mentioned because I tested both inside of docker containers.

Do you develop it in a additional branch or a new repo? Is a roadmap / planned features list available?

**Comment by pwFoo** [comment]
[KeeWeb](https://keeweb.info/) is available as desktop app or just browser based for example.

**Comment by pizzapanther** [comment]
Yes that is already a feature but it's implemented a little differently.  Basically that feature is Neutron Beam and I'll port that over.

Neutron Beam lets you setup remote editing in any environment where you can install a small python program. This makes it a little harder to setup since Cloud9 and Orion are more automated but gives you more flexibility. Also if you wanted to, you could do some custom automation.

Lastly, Neutron Beam gives you to the option to edit locally without an extension or on a internal networked system if you don't trust a public system like Cloud9 or have a requirement where something like that can not be used.
